### PR TITLE
[mesheryctl] Modify the build scripts for support multiple-platform better

### PR DESCRIPTION
**Pull request by Aisuko**
Signed-off-by: Aisuko <urakiny@gmail.com>

#### Description

This PR modify the build scripts for support multiple-platform better

This PR fixes # 

#### Checklist

- [x] [Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin) signed.
- [x] PR includes `[<name of area>]`

### CODE REVIEW

File: mesheryctl/Makefile [hunk 1]
```diff
@@ -2,40 +2,33 @@ make:
...
-	CGO_ENABLED=0 GOARCH=$(ARCH) GOOS=linux GO111MODULE=on go build -ldflags='$(LDFLAGS)' -gcflags='$(GCFLAGS)' -o $(OUTDIR)/$(BINNAME_LINUX) ./cmd/mesheryctl/main.go
 
-.PHONY: $(BINNAME_WINDOWS)
-$(BINNAME_WINDOWS):
-	@echo
-	CGO_ENABLED=0 GOARCH=$(ARCH) GOOS=windows GO111MODULE=on go build -ldflags='$(LDFLAGS)' -gcflags='$(GCFLAGS)' -o $(OUTDIR)/$(BINNAME_WINDOWS) ./cmd/mesheryctl/main.go
+.PHONY: build-multiple
+build-multiple: LDFLAGS += -extldflags "-static"
+build-multiple: $(GOX)
+	GO111MODULE=on CGO_ENABLED=0 $(GOX) -parallel=3 -output="_dist/{{.OS}}-{{.Arch}}/$(BINNAME)" -osarch='$(TARGETS)' $(GOFLAGS) -tags '$(TAGS)' -ldflags '$(LDFLAGS)' ./cmd/mesheryctl/
```
Comment (leecalcote) [inline mesheryctl/Makefile:30]: Has `_dist` been added to .gitignore and `bin` removed?

File: .gitignore [hunk 2]
```diff
@@ -54,4 +54,4 @@ docs/Gemfile.lock
 yarn.lock
 
 # Generated binaries
-**/bin
\ No newline at end of file
+**/_dist
```
Comment (Aisuko) [inline .gitignore:57]: Done

File: mesheryctl/Makefile (hunk 1 above)
Comment (Aisuko) [reply mesheryctl/Makefile:30]: Thanks.

**Comment by Aisuko** [comment]
Any idea we can split the CI to `backend` and `doc/UI`

**Comment by kushthedude** [comment]
> Any ideas with which we can split the CI to backend and doc/UI?

@Aisuko Can you give more context ? Like are you saying to have different jobs for backend & doc/UI and have specific triggers for them ?

**Comment by Aisuko** [comment]
> > Any ideas with which we can split the CI to backend and doc/UI?
> 
> @Aisuko Can you give more context ? Like are you saying to have different jobs for backend & doc/UI and have specific triggers for them ?

Hi @kushthedude , Thanks for tracking, I mean do some conditions for the CI jobs as split the backend and UI/doc. And PR's labels may a good idea.

**Comment by kushthedude** [comment]
> .

@leecalcote This will be accomplished with PR labels as mesheryctl, UI & so on so we can trigger specific jobs.

**Comment by Aisuko** [comment]
If the PR does not include the label, how to trigger the CI manually or trigger the second time

File: mesheryctl/Makefile [hunk 3]
```diff
@@ -2,40 +2,33 @@ make:
 	go build -o mesheryctl cmd/mesheryctl/main.go
 forwin:
 	go build -o mesheryctl.exe cmd/mesheryctl/main.go
```
Comment (leecalcote) [inline mesheryctl/Makefile:4]: @Aisuko @Nikhil-Ladha @nunix @vijcheru Does the `make forwin` work on your Windows machines?

File: .gitignore (hunk 2 above)
Comment (leecalcote) [reply .gitignore:57]: I `_dist` the more common approach as opposed to `bin`?

File: mesheryctl/Makefile (hunk 3 above)
Comment (nunix) [reply mesheryctl/Makefile:4]: just built it from Powershell after I installed gnumake and golang.
Everything went OK as far as I can tell 👍

Comment (Aisuko) [reply mesheryctl/Makefile:4]: `forwin` is work for me.

**Comment by Aisuko** [comment]
Reply to https://github.com/layer5io/meshery/pull/1038#discussion_r433868213 :

Yes, for the multiple platform build scripts.

**Comment by Aisuko** [comment]
Close this PR, because it's so long time to merge and the feature may not necessary for the current version

**Pull request closed without merging**